src/EventListener/TimestampEntityChanges.php

```diff
@@ -29,38 +32,62 @@ public function __construct(Timestamper $timeStamper)
         $this->timeStamper = $timeStamper;
     }
 
-    public function onFlush(OnFlushEventArgs $eventArgs)
+    /**
+     * @param LifecycleEventArgs $args
+     *
+     * @throws Exception
+     */
+    public function postUpdate(LifecycleEventArgs $args)
     {
-        $entityManager = $eventArgs->getEntityManager();
-        $uow = $entityManager->getUnitOfWork();
-        $entities = array_merge(
-            $uow->getScheduledEntityInsertions(),
-            $uow->getScheduledEntityUpdates(),
-            $uow->getScheduledEntityDeletions()
-        );
+        $this->stamp($args->getObject(), new DateTime());
+    }
 
-        foreach ($entities as $entity) {
-            $this->stamp($entity);
-        }
+    /**
+     * @param LifecycleEventArgs $args
+     *
+     * @throws Exception
+     */
+    public function postRemove(LifecycleEventArgs $args)
+    {
+        $this->stamp($args->getObject(), new DateTime());
     }
 
-    protected function stamp($entity)
+    /**
+     * @param LifecycleEventArgs $args
+     *
+     * @throws Exception
+     */
+    public function postPersist(LifecycleEventArgs $args)
+    {
+        $this->stamp($args->getObject(), new DateTime());
+    }
+
+    /**
+     * @param $entity
+     * @param DateTime $timestamp
+     *
+     * @throws Exception
+     */
+    protected function stamp($entity, Datetime $timestamp)
```

Comment:
Is the second argument necessary? Couldn't it just be created here instead since it's always being passed as `now`?

Reply:
i suppose that's right. changed.